commit 887f71b2350f5641be194748c4d0f4e1c5ca1b6a
Author: Boyuan Yang <073plan@gmail.com>
Date:   Thu Jan 25 22:31:39 2018 +0800

    src: engine: ucci.cpp: Workaround the warning about Wwrite-string

diff --git a/src/engine/ucci.cpp b/src/engine/ucci.cpp
index 05c7f34..88e13a9 100644
--- a/src/engine/ucci.cpp
+++ b/src/engine/ucci.cpp
@@ -9,7 +9,9 @@ This part (ucci.h/ucci.cpp only) of codes is NOT published under LGPL, and
 can be used without restriction.
 */
 
-#include <stdio.h>
+#include <cstdio>
+#include <string>
+#include <vector>
 #include "base2.h"
 #include "parse.h"
 #include "pipe.h"
@@ -37,7 +39,13 @@ static bool ParsePos(UcciCommStruct &UcciComm, char *lp) {
     UcciComm.szFenStr = szFen;
   // 然后判断是否是startpos
   } else if (StrEqv(lp, "startpos")) {
-    UcciComm.szFenStr = "rnbakabnr/9/1c5c1/p1p1p1p1p/9/9/P1P1P1P1P/1C5C1/9/RNBAKABNR w - - 0 1";
+    /* trying to workaround warning for ISO C++ about conversion from non-const
+     * to const char*.
+     */
+    const std::string fallback_szFenStr = "rnbakabnr/9/1c5c1/p1p1p1p1p/9/9/P1P1P1P1P/1C5C1/9/RNBAKABNR w - - 0 1";
+    std::vector<char> tmp_szFenStr(fallback_szFenStr.begin(), fallback_szFenStr.end());
+    tmp_szFenStr.push_back('\0');
+    UcciComm.szFenStr = &*tmp_szFenStr.begin();
   // 如果两者都不是，就立即返回
   } else {
     return false;
